File: 1232.check-if-it-is-a-straight-line/solution.kt
class Solution {
    fun checkStraightLine(coordinates: Array<IntArray>): Boolean {
        val x1 = coordinates[0][0]
        val y1 = coordinates[0][1]
        val x2 = coordinates[1][0]
        val y2 = coordinates[1][1]

        fun isOnTheLine(x: Int, y: Int): Boolean {
            return 0 == (y2 - y1) * (x - x1) - (x2 - x1) * (y - y1)
        }

        for (coord in coordinates) {
            if (!isOnTheLine(coord[0], coord[1])) return false
        }
        return true
    }


}

File: 1232.check-if-it-is-a-straight-line/solution.py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        x1,y1 = coordinates[0][0], coordinates[0][1]
        x2,y2 = coordinates[1][0], coordinates[1][1]

        # https://hibikore-tanren.com/line-equation/
        def isOnStraight(x, y):
            return 0 == (y2 - y1) * (x - x1) - (x2 - x1) * (y - y1)

        return all(map(lambda coord: isOnStraight(coord[0], coord[1]), coordinates))
